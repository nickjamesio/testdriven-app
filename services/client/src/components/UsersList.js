import React from "react";
import PropTypes from 'prop-types';

const UsersList = props => {
  return (
    <div>
      {props.users.map(user => {
        return (
          <h4 key={user.id} className="box title is-4">
            {user.username}
          </h4>
        );
      })}
    </div>
  );
};

UsersList.propTypes = {
    users: PropTypes.array.isRequired
}

export default UsersList;
