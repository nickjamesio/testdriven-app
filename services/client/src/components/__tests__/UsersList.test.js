import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UsersList from '../UsersList';

const users = [
  {
    'active': true,
    'email': 'nickjames@gmail.com',
    'id': 1,
    'username': 'nick'
  },
  {
    'active': true,
    'email': 'nick@james.org',
    'id': 2,
    'username': 'nick_james'
  }
];

test('UsersList renders properly', () => {
  const wrapper = shallow(<UsersList users={users}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('nick');
});

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});