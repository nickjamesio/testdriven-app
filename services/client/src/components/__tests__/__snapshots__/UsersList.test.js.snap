// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    nick
  </h4>
  <h4
    className="box title is-4"
  >
    nick_james
  </h4>
</div>
`;
